src/server.js

```diff
@@ -35,10 +37,22 @@ export default function start(
 
 	const baseRoutes = getRoutes(renderRequestMap);
 	const finalRoutes = [...routes, ...baseRoutes];
+	let tls = null;
+	if (
+		appConfig.app_server.protocol === 'https' &&
+		appConfig.app_server.key_file &&
+		appConfig.app_server.crt_file
```

Comment:
as part of validation, we should make sure that `key_file` and `crt_file` are populated when `protocol` is https - that's a pretty clear error condition. Then, we'll be able to assume that if `protocol === 'https'`, we're good to go.